### ISSUE TITLE
Handle UTF16 lone surrogates

### ISSUE THREAD

**Issue by bmeck** [oxigraph/json-event-parser#1]
JSON uses 2 byte hex encoding for unicode escapes. This can cause lone surrogate problems in UTF-8 encodings that need 4 bytes ( see https://unicode-example-characters.glitch.me/ ). For example:

🔥 can be encoded as "\ud83d\udd25".
🚸🚯 can be encoded as "\ud83d\udeb8\ud83d\udeaf".

Currently attempting to decode these results in errors: 

```rs
    let json = b"\"\\ud83d\\udd25\"";
    let mut reader = JsonReader::from_reader(Cursor::new(json));

    let mut buffer = Vec::new();
    reader.read_event(&mut buffer)?;
```

```console
panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "Invalid encoded unicode code point" }', src/main.rs:86:15
```

**Comment by Tpt** [comment]
Thank you for this bug report! I should have indeed added support for surrogate pairs. I have just [implemented them](https://github.com/oxigraph/json-event-parser/commit/63f99048d29c00d1dd74379a796d7f66991fac45) and released a v0.1.1 version. Your test now passes. Thank you again!

**Comment by bmeck** [comment]
Thank you so much!
